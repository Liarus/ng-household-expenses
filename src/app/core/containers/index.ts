export * from './app';
export * from './page-not-found';
