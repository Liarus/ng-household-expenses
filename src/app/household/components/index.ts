export * from './household-list';
export * from './household-dialog';
export * from './household-tiles';
export * from './household-search';
