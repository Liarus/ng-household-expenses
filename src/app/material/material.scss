.mat-bottom-sheet-container {
  padding: 0 !important;
}

.mat-table {
  mat-row, mat-header-row, mat-footer-row {
    padding-left: 24px;
    padding-right: 24px;
  }
  
  mat-cell:first-child, mat-footer-cell:first-child, mat-header-cell:first-child {
    padding-left: 0;
  }
  
  mat-cell:last-child, mat-footer-cell:last-child, mat-header-cell:last-child {
    padding-right: 0;
  }
}

mat-expansion-panel {
  mat-expansion-panel-header {
    height: 48px; // height may be different for you
  }
  .mat-expansion-panel-content {
    height: 0;
  }
  &.mat-expanded {
    mat-expansion-panel-header {
      height: 64px; // height may be different for you
    }
    .mat-expansion-panel-content {
      height: auto;
    }
  }
}