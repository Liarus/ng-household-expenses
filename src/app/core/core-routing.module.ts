import { Routes, RouterModule } from '@angular/router';

import { LayoutComponent } from './containers';
import { NgModule } from '@angular/core';

export const routes: Routes = [
    {
        path: 'root',
        component: LayoutComponent
    }
];

@NgModule({
    imports: [RouterModule.forRoot(routes)],
    exports: [RouterModule]
  })
export class CoreRoutingModule { }
