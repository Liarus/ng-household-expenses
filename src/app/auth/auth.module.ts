import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { ReactiveFormsModule } from '@angular/forms';
import { StoreModule } from '@ngrx/store';
import { EffectsModule } from '@ngrx/effects';

import { AuthEffects } from './store/effects/auth.effects';
import { reducers } from './store/reducers';
import { AuthRoutingModule } from './auth-routing.module';
import { LoginPageComponent } from './containers';
import { LoginFormComponent } from './components';
import { MaterialModule } from '../material/material.module';

export const COMPONENTS = [
  LoginPageComponent,
  LoginFormComponent
];

@NgModule({
  imports: [
    CommonModule,
    ReactiveFormsModule,
    AuthRoutingModule,
    MaterialModule,
    StoreModule.forFeature('auth', reducers),
    EffectsModule.forFeature([AuthEffects]),
  ],
  declarations: COMPONENTS,
})
export class AuthModule {}
