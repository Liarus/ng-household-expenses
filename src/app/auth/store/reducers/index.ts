import { ActionReducerMap, createFeatureSelector, createSelector } from '@ngrx/store';

import * as fromAuth from './auth.reducer';
import * as fromRoot from '../../../store/reducers';

export interface AuthState {
    status: fromAuth.State;
}

export interface State extends fromRoot.State {
    auth: AuthState;
}

export const reducers: ActionReducerMap<AuthState> = {
    status: fromAuth.reducer
};

export const getAuthState = createFeatureSelector<AuthState>('auth');

export const getStatusState = createSelector(
    getAuthState,
    (state: AuthState) => state.status
);

export const getLoggedIn = createSelector(
    getStatusState,
    fromAuth.getLoggedIn
);

export const getLoading = createSelector(
    getStatusState,
    fromAuth.getLoading
);

export const getErrorMessage = createSelector(
    getStatusState,
    fromAuth.getErrorMessage
);

export const getLoggedUser = createSelector(
    getStatusState,
    fromAuth.getLoggedUser
);

export const getAccessToken = createSelector(
    getStatusState,
    fromAuth.getAccessToken
);
