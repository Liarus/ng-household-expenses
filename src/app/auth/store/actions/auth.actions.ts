import { Action } from '@ngrx/store';

import { SignInRequest } from '../../models/requests/signInRequest.model';
import { ErrorMessage } from '../../../shared/models/errorMessage.model';
import { User } from '../../models/user.model';

export enum AuthActionTypes {
    Login = '[Auth] Login',
    Logout = '[Auth] Logout',
    LoginSuccess = '[Auth API] Login Success',
    LoginFailure = '[Auth API] Login Failure',
    LoginRedirect = '[Auth API] Login Redirect',
}

export class Login implements Action {
    readonly type = AuthActionTypes.Login;

    constructor(public payload: SignInRequest) {}
}

export class Logout implements Action {
    readonly type = AuthActionTypes.Logout;
}

export class LoginSuccess implements Action {
    readonly type = AuthActionTypes.LoginSuccess;

    constructor(public payload: {
        user: User,
        accessToken: string
    }) {}
}

export class LoginFailure implements Action {
    readonly type = AuthActionTypes.LoginFailure;

    constructor(public payload: ErrorMessage) {}
}

export class LoginRedirect implements Action {
    readonly type = AuthActionTypes.LoginRedirect;
}

export type AuthActionsUnion =
    | Login
    | Logout
    | LoginSuccess
    | LoginFailure
    | LoginRedirect;
