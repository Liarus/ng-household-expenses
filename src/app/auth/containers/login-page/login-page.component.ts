import { Component, OnInit, ChangeDetectionStrategy } from '@angular/core';
import { Store, select } from '@ngrx/store';

import * as fromAuth from '../../store/reducers';
import * as AuthActions from '../../store/actions/auth.actions';
import { LoginRequest } from '../../models/requests/loginRequest.model';

@Component({
  selector: 'app-login-page',
  changeDetection: ChangeDetectionStrategy.OnPush,
  template: `
    <app-login-form
      [pending]="isLoading$ | async"
      (submitted)="onSubmit($event)"
    ></app-login-form>
  `
})
export class LoginPageComponent implements OnInit {

  isLoading$ = this.store.pipe(select(fromAuth.getLoading));

  constructor(private store: Store<fromAuth.State>) { }

  ngOnInit() {
  }

  onSubmit($event: LoginRequest) {
    this.store.dispatch(new AuthActions.Login($event));
  }
}
