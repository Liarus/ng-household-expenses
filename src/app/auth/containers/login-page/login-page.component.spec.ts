/// <reference types="jest" />
import { async, ComponentFixture, TestBed } from '@angular/core/testing';
import { NoopAnimationsModule } from '@angular/platform-browser/animations';
import { ReactiveFormsModule } from '@angular/forms';
import { StoreModule, Store } from '@ngrx/store';

import { LoginPageComponent } from './login-page.component';
import { LoginFormComponent } from '../../components';
import { MaterialModule } from '../../../material/material.module';
import { MockStore } from '../../../shared/tests/mockStore';
import * as AuthActions from '../../store/actions/auth.actions';
import { LoginRequest } from '../../models/requests/loginRequest.model';

describe('LoginPageComponent', () => {
  let component: LoginPageComponent;
  let fixture: ComponentFixture<LoginPageComponent>;

  let store: MockStore<any>;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      imports: [
        MaterialModule,
        NoopAnimationsModule,
        ReactiveFormsModule,
        StoreModule.forRoot({})
      ],
      declarations: [
        LoginPageComponent,
        LoginFormComponent
      ],
      providers: [{ provide: Store, useClass: MockStore }]
    })
      .compileComponents();
  }));

  beforeEach(() => {
    store = TestBed.get(Store);
    store.setState({
      auth: {
        status: {
          loading: false
        }
      }
    });
    fixture = TestBed.createComponent(LoginPageComponent);
    component = fixture.componentInstance;
    spyOn(store, 'dispatch');
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should match snapshot', () => {
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  });

  it('should dispatch Login action', () => {
    const credentials: LoginRequest = {
      username: 'user@test.comn',
      password: 'test'
    };
    const expected = new AuthActions.Login(credentials);

    component.onSubmit(credentials);

    expect(store.dispatch).toHaveBeenLastCalledWith(expected);
  });
});
