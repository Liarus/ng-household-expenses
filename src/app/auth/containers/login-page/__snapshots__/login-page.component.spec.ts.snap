// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginPageComponent should match snapshot 1`] = `
<app-login-page
  isLoading$={[Function Store]}
  store={[Function MockStore]}
>
  <app-login-form
    ng-reflect-pending="false"
  >
    <mat-card
      class="mat-card"
    >
      <mat-card-title
        class="mat-card-title"
      >
        Login
      </mat-card-title>
      <mat-card-content
        class="mat-card-content"
      >
        <form
          class="ng-untouched ng-pristine ng-invalid"
          ng-reflect-form="[object Object]"
          novalidate=""
        >
          <p>
            <mat-form-field
              class="mat-form-field ng-tns-c4-0 mat-primary mat-form-field-type-mat-input mat-form-field-appearance-legacy mat-form-field-can-float mat-form-field-hide-placeholder ng-untouched ng-pristine ng-invalid _mat-animation-noopable"
            >
              <div
                class="mat-form-field-wrapper"
              >
                <div
                  class="mat-form-field-flex"
                >
                  
                  
                  <div
                    class="mat-form-field-infix"
                  >
                    <input
                      aria-invalid="false"
                      aria-required="false"
                      class="mat-input-element mat-form-field-autofill-control cdk-text-field-autofill-monitored ng-untouched ng-pristine ng-invalid"
                      formcontrolname="email"
                      id="mat-input-2"
                      matinput=""
                      ng-reflect-name="email"
                      ng-reflect-placeholder="Email"
                      ng-reflect-type="text"
                      placeholder="Email"
                      type="text"
                    />
                    <span
                      class="mat-form-field-label-wrapper"
                    >
                      
                      <label
                        aria-owns="mat-input-2"
                        class="mat-form-field-label ng-tns-c4-0 mat-empty mat-form-field-empty ng-star-inserted"
                        for="mat-input-2"
                        id="mat-form-field-label-5"
                        ng-reflect-disabled="true"
                        ng-reflect-ng-switch="false"
                      >
                        
                        
                        Email
                        
                        
                      </label>
                    </span>
                  </div>
                  
                </div>
                
                <div
                  class="mat-form-field-underline ng-tns-c4-0 ng-star-inserted"
                >
                  <span
                    class="mat-form-field-ripple"
                  />
                </div>
                <div
                  class="mat-form-field-subscript-wrapper"
                  ng-reflect-ng-switch="hint"
                >
                  
                  
                  <div
                    class="mat-form-field-hint-wrapper ng-tns-c4-0 ng-trigger ng-trigger-transitionMessages ng-star-inserted"
                    style="opacity:1;transform:translateY(0%);0:opacity;1:transform;"
                  >
                    
                    <div
                      class="mat-form-field-hint-spacer"
                    />
                  </div>
                </div>
              </div>
            </mat-form-field>
          </p>
          <p>
            <mat-form-field
              class="mat-form-field ng-tns-c4-1 mat-primary mat-form-field-type-mat-input mat-form-field-appearance-legacy mat-form-field-can-float mat-form-field-hide-placeholder ng-untouched ng-pristine ng-invalid _mat-animation-noopable"
            >
              <div
                class="mat-form-field-wrapper"
              >
                <div
                  class="mat-form-field-flex"
                >
                  
                  
                  <div
                    class="mat-form-field-infix"
                  >
                    <input
                      aria-invalid="false"
                      aria-required="false"
                      class="mat-input-element mat-form-field-autofill-control cdk-text-field-autofill-monitored ng-untouched ng-pristine ng-invalid"
                      formcontrolname="password"
                      id="mat-input-3"
                      matinput=""
                      ng-reflect-name="password"
                      ng-reflect-placeholder="Password"
                      ng-reflect-type="password"
                      placeholder="Password"
                      type="password"
                    />
                    <span
                      class="mat-form-field-label-wrapper"
                    >
                      
                      <label
                        aria-owns="mat-input-3"
                        class="mat-form-field-label ng-tns-c4-1 mat-empty mat-form-field-empty ng-star-inserted"
                        for="mat-input-3"
                        id="mat-form-field-label-7"
                        ng-reflect-disabled="true"
                        ng-reflect-ng-switch="false"
                      >
                        
                        
                        Password
                        
                        
                      </label>
                    </span>
                  </div>
                  
                </div>
                
                <div
                  class="mat-form-field-underline ng-tns-c4-1 ng-star-inserted"
                >
                  <span
                    class="mat-form-field-ripple"
                  />
                </div>
                <div
                  class="mat-form-field-subscript-wrapper"
                  ng-reflect-ng-switch="hint"
                >
                  
                  
                  <div
                    class="mat-form-field-hint-wrapper ng-tns-c4-1 ng-trigger ng-trigger-transitionMessages ng-star-inserted"
                    style="opacity:1;transform:translateY(0%);0:opacity;1:transform;"
                  >
                    
                    <div
                      class="mat-form-field-hint-spacer"
                    />
                  </div>
                </div>
              </div>
            </mat-form-field>
          </p>
          <p
            class="loginButtons"
          >
            <button
              class="mat-button _mat-animation-noopable"
              mat-button=""
              type="submit"
            >
              <span
                class="mat-button-wrapper"
              >
                Login
              </span>
              <div
                class="mat-button-ripple mat-ripple"
                matripple=""
                ng-reflect-centered="false"
                ng-reflect-disabled="false"
                ng-reflect-trigger="[object HTMLButtonElement]"
              />
              <div
                class="mat-button-focus-overlay"
              />
            </button>
          </p>
        </form>
      </mat-card-content>
    </mat-card>
  </app-login-form>
</app-login-page>
`;
